azerothcore/azerothcore-wotlk
# [Core/Scripts]: Add locale support to Autobroadcast

## Changes Proposed:
- Move autobroadcast to it's own file.
- Add support for announcing text for different client locales.
- Move WorldTextBuilder class definition to World.h Nothing really changes except we can now access it outside of World.cpp But we still need to do it through the Acore namespace.

## Issues Addressed:

- Closes Autobroadcast part of https://github.com/azerothcore/azerothcore-wotlk/issues/2856

## SOURCE:


## Tests Performed:

- Tested with en_US client and the rest of locales were viewed in the console and they seemed to display.


## How to Test the Changes:


1. Add your custom locale message in the autobroadcast table according to your client's locale.
2. Login and see if you get the message.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
